### PR TITLE
Tweak: Add a VideoPress integration for self-hosted videos [ED-9068]

### LINKED ISSUE #20617: 🔀 🚀 Feature Request: VideoPress integration for self-hosted videos (PR #20569)
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

When using VideoPress, video widget with self-hosted media are not rendered correctly.
The self-hosted link will grab the video URL when it should use the streaming playlist and the VideoPress player.
We also can't paste VideoPress URLs as external URLs.

### Describe the solution you'd like

When using a video widget and selecting a VideoPress video from the media library, the VideoPress player should be used instead of a video tag with the video URL.
When using a video widget and pasting a VideoPress URL in self-hosted "External link", the result should be the same.

### Describe alternatives you've considered

The only solution right now is to use the Gutenberg block editor.

### Additional context

A PR with proposed changes is available here #20569

### PR DESCRIPTION
**Pull request by thedebian**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Use the VideoPress player for self-hosted VideoPress videos.
* Use the VideoPress player for self-hosted VideoPress external URLs.
* Fix a PHP warning when "lazy_load" is not in the embed options.

## Description

This PR creates a new video type for VideoPress videos in the Video widget.

The benefit to the user of using the VideoPress player over the default `<video>` tag is it is able to take advantage of all of the features that VideoPress provides such as using a highly compatible transcoded video file for playback on many types of devices, adaptive streaming based on device screen size and available bandwidth and the ability to authenticate private videos.

## Test instructions
This PR can be tested by following these steps:

* Apply this PR
* Create a new page or post
* Insert a video widget
* Select "VideoPress" as source
* Select "External URL"
* Set the URL to a public VideoPress URL, eg. `https://video.wordpress.com/embed/Ygmx4akX`
* ✅ The video should be properly loaded in the VideoPress player
* Test the same video GUID with different valid base URLs like:
    * `https://video.wordpress.com/v/Ygmx4akX`
    * `http://video.wordpress.com/v/Ygmx4akX`
    * `https://videopress.com/embed/Ygmx4akX`
    * `https://videopress.com/v/Ygmx4akX`
    * `http://videopress.com/v/Ygmx4akX`
    * `https://www.videopress.com/v/Ygmx4akX`
    * `videopress.com/v/Ygmx4akX`
    * `https://videos.files.wordpress.com/Ygmx4akX/red-line-1_mp4_std.mp4`
* ✅ All these URLs should load the VideoPress player with the correct video
* For the next step, two possibilites:
    * On a site with an active VideoPress subscription
    * Or Install Jetpack VideoPress and choose the free plan and make sure VideoPress is enabled
* Go to your media library and upload a video
* Copy the video URL starting with "videos.files.wordpress.com" or "video.wordpress.com" or "videopress.com"
* Go back to your post/page or create a new one
* Insert a video widget and select "VideoPress"
* Select the video you uploaded on VideoPress from your media library
* ✅ The VideoPress player should be displayed
* Select "external URL" and paste the URL you previously copied
* ✅ The VideoPress player should be displayed and the proper video should load.
* Make your video private by switching the "visibility" toggle in the media library
* Go back to your post/page
* ✅ The video should still work
* Publish your post/page
* Go to the frontend
* ✅ The video should still work
* Open an incognito tab and go to your page/post frontend
* ✅ The video should be displayed as private

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [x] Docs have been added / updated (for bug fixes / features)

Fix #20617

### CODE REVIEW

File: includes/widgets/video.php [hunk 1]
```diff
@@ -1262,9 +1262,79 @@ private function render_hosted_video() {
 		}
 
 		$video_params = $this->get_hosted_params();
-		/* Sometimes the video url is base64, therefore we use `esc_attr` in `src`. */
-		?>
-		<video class="elementor-video" src="<?php echo esc_attr( $video_url ); ?>" <?php Utils::print_html_attributes( $video_params ); ?>></video>
-		<?php
+
+		if ( ! $this->render_hosted_videopress_video( $video_url, $video_params ) ) {
```
Comment (matipojo) [inline includes/widgets/video.php:1266]: Please add it as a video type (https://github.com/elementor/elementor/pull/20569/files#diff-075263b807b9f3432f51fac037c717bcea04e7884b5174b9d8a2f9da0ea3f79fL117)
then you can add a specific control like `videopress_url`

**Review by matipojo — DISMISSED** [review]
Hi @thedebian thank for the PR, can you please make the videopress video type to act like the other providers? (see the comment above)

**Comment by thedebian** [comment]
> Hi @thedebian thank for the PR, can you please make the videopress video type to act like the other providers? (see the comment above)

Hi @matipojo !

The rationale behind this implementation was that VideoPress videos are technically seen as self-hosted on the user's site, as they're integrated in the media library.
This is causing problems for most (if not all) users using Elementor with VideoPress (either because they're on a WordPress.com plan or a Jetpack VideoPress plan), because they'll select the video and the direct video link will be used instead of the VideoPress player, so they won't have adaptive streaming, etc...
We might also block direct linking at some point, as specified in our terms of service, hence our effort towards integrating the service properly in popular WordPress plugins, like Elementor, so users won't just get broken videos everywhere.

I can absolutely add VideoPress as a provider though (in fact, that's what I did first, before realizing it would be an almost carbon-copy of the "self-hosted" provider), maybe we (or I) should find a way to filter out those videos when using the "self-hosted" provider then, so users won't use the "wrong" one ?

**Comment by thedebian** [comment]
Hi again @matipojo ,
I updated the PR so VideoPress is now a video type in the video widget.
In order for video URLs to be properly matched, I also had to make a slight update in embed.php, so multiple patterns could be used for a single provider (the other solution would be to make a gigantic pattern that'd be hard to maintain).
Don't hesitate to come back to me for any update !

**Comment by thedebian** [comment]
Hey there @matipojo, I'm pinging to check if there's any update on this PR ?

**Comment by avivu** [comment]
Hello @thedebian 
Can you please update the base branch so we can further check this PR and implementation?
Thank you!

**Comment by thedebian** [comment]
Hi @avivu I changed the base to main 👍

**Review by arielk — APPROVED** [review]

**Pull request merged**